packages/@expo/cli/src/start/platforms/PlatformManager.ts

```diff
@@ -49,14 +52,31 @@ export class PlatformManager<
     throw new UnimplementedError();
   }
 
-  protected async openProjectInExpoGoAsync(
+  protected async openProjectInExpoGoOrInterstitialPageAsync(
     resolveSettings: Partial<IResolveDeviceProps> = {}
   ): Promise<{ url: string }> {
-    const url = this.props.getExpoGoUrl();
+    const interstitialPageUrl = this.props.getInterstitialPageUrl();
+    const expoGoUrl = this.props.getExpoGoUrl();
+    const deviceManager = await this.props.resolveDeviceAsync(resolveSettings);
+
+    let url = expoGoUrl;
+    if (interstitialPageUrl) {
```

Comment:
I'd prefer if this was the `shouldUseInterstitialPage()` and the URL logic stays combined. The model here is spiritually similar to universal links on iOS: user open apps (Expo Go) -> more than one target -> disambiguate. But the intention should always be "open in dev client" or "open in Expo Go" instead of "open in interstitial".